### PR TITLE
debug: Add renderdoc capture hotkey

### PR DESCRIPTION
**Pull request by GPUCode**
Debugging the texture or buffer cache is often difficult because certain resources are uploaded at specific points and then used in many subsequent frames. Capturing these specific frames in renderdoc is possible but cumbersome. This is my attempt at a solution to this problem.

The PR uses the renderdoc C API and adds a hotkey to start/stop frame captures. This way multiple frames can be recorded and written into the same large capture. Then using the resource inspector the desired frame can be found and analyzed. I've only enabled this if graphics debugging is used as it's a strictly debugging feature.

### CODE REVIEW

**Comment by liamwhite** [comment]
It would be nice to not have this connected to the validation layers as they make debugging with renderdoc painfully slow.

**Comment by GPUCode** [comment]
Should I add an addition setting for this, or make it always available?

**Comment by liamwhite** [comment]
Whatever you think is best, I have no strong preference

**Comment by GPUCode** [comment]
Added new setting
εικόνα

File: src/core/tools/renderdoc.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#else
+#include <dlfcn.h>
+#endif
+
+namespace Tools {
+
+RenderdocAPI::RenderdocAPI() {
+#ifdef WIN32
+    if (HMODULE mod = GetModuleHandleA("renderdoc.dll")) {
+        const auto RENDERDOC_GetAPI = (pRENDERDOC_GetAPI)GetProcAddress(mod, "RENDERDOC_GetAPI");
```
Comment (FearlessTobi) [inline src/core/tools/renderdoc.cpp:20]: should this be a static cast?

**Comment by vonchenplus** [comment]
The performance of continuous capture is very poor and other options may need to be investigated.

**Comment by vonchenplus** [comment]
Often, successive captures are of little use, consider increasing the capture interval.

**Comment by GPUCode** [comment]
Having a capture interval risks not capturing the frame that contains the resource upload, which makes the feature not useful. This is a hard tradeoff as having a small interval, there won't be much performance improvement, while having a larger one will result in many captures and reduce the chances one of them has the wanted frame.

File: src/core/tools/renderdoc.cpp (hunk 1 above)
Comment (liamwhite) [reply src/core/tools/renderdoc.cpp:20]: reinterpret, probably

File: src/core/tools/renderdoc.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        const s32 ret = RENDERDOC_GetAPI(eRENDERDOC_API_Version_1_6_0, (void**)&rdoc_api);
+        ASSERT(ret == 1);
```
Comment (liamwhite) [inline src/core/tools/renderdoc.cpp:22]: needs common/assert.h

**Comment by liamwhite** [comment]
@GPUCode needs rebase, also see comment above

**Review by liamwhite — APPROVED** [review]

**Pull request merged**